src/hardware/dac.rs

```diff
@@ -39,6 +41,38 @@ pub enum Error {
     Bounds,
 }
 
+/// A type representing a DAC sample.
+#[derive(Copy, Clone, Debug, Format)]
+pub struct DacCode(u32);
+impl DacCode {
+    // DAC constants
+    const MAX_DAC_WORD: i32 = 1 << 20; // maximum DAC dataword (exclusive) plus 2 bit due to interface alignment
+    const VREF_DAC: f32 = 3.0; // DAC reference voltage
+    pub const MAX_CURRENT: f32 = 3.0 - (6.0 / (1 << 18) as f32); // 3.0 A minus one DAC LSB in eqivalent Ampere.
```

Comment:
How would you implement this?

Reply:
Solve the expressions in L59 and L60 for current given MAX_DAC_CODE - 1.